docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -231,6 +231,50 @@ var finder = new AboutToExpireSecretFinder(TimeSpan.FromDays(2), secretClient);
 
 This approach is useful when you would like to consolidate the dependency creation and share the client between multiple consuming classes.
 
+## Explore the extension methods in Azure Resource Manager SDKs
+
+For fundamental concepts of Azure Resource Manager SDKs, see [Key Concepts of Azure Resource Manager SDKs](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/resourcemanager/Azure.ResourceManager#key-concepts).
+
+In Azure Resource Manager SDKs, we design the clients (resources and collections) to have their hierarchical structures to emphasize their relationship between each other. To achieve that, extension methods are widely used because the parent resource of one resource might not exist in the same SDK library.
+
+For instance, in `Azure.ResourceManager.Compute` namespace, there is a resource `VirtualMachineResource` which stands for a virtual machine resource on Azure, and its parent is `ResourceGroupResource` in `Azure.ResourceManager` namespace, and we could get the `VirtualMachineCollection` on the `ResourceGroupResource` in this way:
+
+:::code language="csharp" source="snippets/unit-testing/ResourceManager/ResourceManagerCodeStructure.cs" id="ParentOfVMIsRG" :::
+
+To make sure these extension methods are able to be mocked or unit tested, these methods are always implemented in this way:
+
+:::code language="csharp" source="snippets/unit-testing/ResourceManager/ComputeExtensions.cs" id="HowExtensionMethodsAreImplemented" :::
```

Comment:
Wrap longer lines of code to prevent horizontal scrolling in the doc. Here's how the code currently displays:

![image](https://github.com/dotnet/docs/assets/10702007/ad5cc43d-d1c7-4426-8f74-456b6555bdbc)

Reply:
does this width vary on different monitor or resolution?
On my preview, some of them need to be wrapped but not as this much.
For instance, the `public static VirtualMachineCollection GetVirtualMachines` method could fit fine on my preview.

Do we have a recommended number for how many characters we could have on one line?

BTW I wrapped some lines based on the preview on my side.

Reply:
And when we have to wrap a long line of a method declaration, where should we put the `(`? On the first line or the second?
I am currently doing this:
```csharp
    internal ComputeResourceGroupMockingExtension(
        ArmClient client, ResourceIdentifier id) : base(client, id)
    { }
```
should I do this?
```csharp
    internal ComputeResourceGroupMockingExtension
        (ArmClient client, ResourceIdentifier id) : base(client, id)
    { }
```

Reply:
The recommended number of chars is 85. See https://review.learn.microsoft.com/help/platform/code-in-docs?branch=main#horizontal-scroll-bars. My preference is to put `(` on the first line.

Reply:
These details now are hidden and no longer show up on the page therefore I believe it is fine. I will pay attention to other places to see if we have horizontal scoll bars